moveit_core/kinematics_base/src/kinematics_base.cpp

```diff
@@ -92,6 +92,26 @@ bool KinematicsBase::initialize(const rclcpp::Node::SharedPtr& /*node*/,
   return false;
 }
 
+bool KinematicsBase::initialize(const node_interface::NodeInterfaceSharedPtr& node, const moveit::core::RobotModel& robot_model, const std::string& group_name,
+                                const std::string& base_frame, const std::vector<std::string>& tip_frames,
+                                double search_discretization)
+{
+
+
+    if (node->has_lifecycle_node_){
+        RCLCPP_ERROR(LOGGER,
+                     "IK plugin for group '%s' relies on deprecated API. "
+                     "Please implement initialize(rclcpp::Node::SharedPtr, RobotModel, ...).",
+                     group_name.c_str());
+    } else{
+        // try rclcpp node interface
+        return initialize(node->get_rcl_node(), robot_model, group_name, base_frame, tip_frames,
+                          search_discretization);
```

Comment:
I'm confused by this, if the node interface is the lifecycle one, it won't initalize?

Reply:
Becuase I chnaged KinematicsPluginLoader::KinematicsLoaderImpl to call KinematicsBase::initialize with NodeInterfaceSharedPtr instead of rclcpp::Node::SharedPtr, IK plugins that use rclcpp::Node::SharedPt would now never get loaded. If this ocde is hit, it means the user did not load a plugin with the NodeInterfaceSharedPtr interface. So I need to call KinematicsBase::initialize with rclcpp::Node::SharedPtr in case that interface is being used.